java/org/contikios/cooja/plugins/Mobility.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2009, Swedish Institute of Computer Science.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the Institute nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE INSTITUTE AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE INSTITUTE OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ */
+
+package org.contikios.cooja.plugins;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import javax.swing.JFileChooser;
+import javax.swing.JScrollPane;
+
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+
+import org.contikios.cooja.ClassDescription;
+import org.contikios.cooja.Cooja;
+import org.contikios.cooja.Mote;
+import org.contikios.cooja.PluginType;
+import org.contikios.cooja.Simulation;
+import org.contikios.cooja.TimeEvent;
+import org.contikios.cooja.VisPlugin;
+import org.contikios.cooja.dialogs.MessageListUI;
+import org.contikios.cooja.interfaces.Position;
+import org.contikios.cooja.util.StringUtils;
+
+import org.jdom.Element;
+
+
+@ClassDescription("Mobility")
+@PluginType(PluginType.SIM_PLUGIN)
+public class Mobility extends VisPlugin {
+  private static Logger logger = LogManager.getLogger(Mobility.class);
+
+  private static final boolean QUIET = false;
+
+  private final boolean WRAP_MOVES = true; /* Wrap around loaded moves forever */
+  
```

Comment:
Trailing whitespace on this line and a few more places.

I think modern git will show the trailing whitespace in red if you do `git show` on a commit.

Enabling the IDE/Editor to always remove all trailing whitespace when saving is a bad idea, but you can `M-x whitespace-cleanup` in emacs to remove trailing whitespace on a specific buffer.

Reply:
I have now removed the trailing whitespaces. Thanks